src/Microsoft.AspNet.Server.Kestrel/Networking/Libuv.cs

```diff
@@ -199,6 +200,9 @@ public void async_init(UvLoopHandle loop, UvAsyncHandle handle, uv_async_cb cb)
         protected Func<UvAsyncHandle, int> _uv_async_send;
         public void async_send(UvAsyncHandle handle)
         {
+            // Can't Assert with .Validate as that checks threadId
+            // and this function is to post to correct thread.
+            Debug.Assert(!handle.IsInvalid, "Handle is invalid");
```

Comment:
If you put just this change in and run ./build in Windows; debug when the Assert pops up you can clearly see what's going on.